src/hooks/useUploadImage.ts

```diff
@@ -0,0 +1,28 @@
+import {Image} from 'react-native-compressor';
+import {composeImageMeta} from '../utils/string/file';
+import {uploadPhoto} from '../service/file';
+
+const useUploadImage = () => {
+  const uploadPhotoImage = async (pathImg: string) => {
+    try {
+      const compressionResult = await Image.compress(pathImg, {
+        compressionMethod: 'auto'
+      });
+      console.log('kaki', compressionResult);
+      const asset = new FormData();
+      asset.append('photo', composeImageMeta(compressionResult));
+
+      const responseUpload = await uploadPhoto(asset);
+      console.log({responseUpload}, 'lahap');
+      return responseUpload;
+    } catch (e) {
+      console.log(e);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling here is not sufficient. Simply logging the error does not provide any feedback to the user or handle the failure case in the application logic. Consider throwing the error or returning a specific value to indicate an error occurred.

```diff
- console.log(e);
+ throw e;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->